<div style="width:100%;box-sizing:border-box;padding:24px 26px;border-radius:18px;background:linear-gradient(135deg,#EEF2FF 0%,#F5F3FF 55%,#ECFEFF 100%);border:1px solid #C7D2FE;box-shadow:0 6px 18px rgba(79,70,229,0.08);font-family:Arial,sans-serif;text-align:center;">
<div style="font-size:13px;font-weight:700;color:#6366F1;letter-spacing:0.5px;margin-bottom:8px;">🧪 HANDS-ON LAB</div>
<div style="font-size:26px;font-weight:700;color:#312E81;">Week 7 · Day 4 · Hands-On Lab</div>
<div style="font-size:15px;color:#64748B;margin-top:8px;">Using a Pre-trained Transformer</div>
</div>

<div style="width:100%;box-sizing:border-box;padding:17px 22px;border-radius:15px;background:linear-gradient(135deg,#FFF7ED,#FFF1F2);border:1px solid #FED7AA;box-shadow:0 4px 12px rgba(249,115,22,0.06);font-family:Arial,sans-serif;">
<div style="font-size:21px;font-weight:700;color:#C2410C;">🔹 Step 1 · Pre-trained Transformer</div>
<div style="font-size:13px;color:#6B7280;margin-top:6px;">Running a ready-to-use Transformer model with Hugging Face</div>
</div>

In [ ]:
from transformers import pipeline

classifier = pipeline("sentiment-analysis")

result = classifier("This internship is excellent!")

result

[transformers] No model was supplied, defaulted to distilbert/distilbert-base-uncased-finetuned-sst-2-english and revision 714eb0f.
Using a pipeline without specifying a model name and revision in production is not recommended.
Loading weights: 100%|██████████| 104/104 [00:00<00:00, 4894.77it/s]


[{'label': 'POSITIVE', 'score': 0.9998610019683838}]

: 

<div style="width:100%;box-sizing:border-box;padding:20px 22px;border-radius:15px;background:#FFFFFF;border:1px solid #99F6E4;box-shadow:0 4px 12px rgba(20,184,166,0.05);font-family:Arial,sans-serif;">
<div style="font-size:19px;font-weight:700;color:#0F766E;margin-bottom:10px;">📊 Result Interpretation</div>
<div style="font-size:14px;line-height:1.75;color:#4B5563;">
The pre-trained Transformer classified the sentence as <b>POSITIVE</b>. The score of <b>0.999861</b> represents the model's confidence in this prediction.
<br><br>
This demonstrates how a pre-trained Transformer can perform a useful NLP task without training the model from scratch.
</div>
</div>

<div style="width:100%;box-sizing:border-box;padding:20px 22px;border-radius:15px;background:linear-gradient(135deg,#F0FDFA,#ECFEFF);border:1px solid #99F6E4;box-shadow:0 4px 12px rgba(20,184,166,0.06);font-family:'Segoe UI',Arial,sans-serif;">
<div style="font-size:19px;font-weight:900;color:#0F766E;margin-bottom:10px;">🔹 Step 2 · Applying the Transformer to Project Data</div>
<div style="font-size:14px;line-height:1.8;color:#4F5D63;">
The project is based on sequential ECG data rather than text. Therefore, the pre-trained NLP sentiment-analysis pipeline used in Step 1 is not directly applicable to the ECG data. Since the Day 4 instruction specifies applying the pre-trained Transformer to project data when the project is text-based, no direct Transformer-versus-LSTM metric comparison is performed here. The LSTM remains the valid model for the current ECG classification task.
</div>
</div>

<div style="width:100%;box-sizing:border-box;padding:20px 22px;border-radius:15px;background:linear-gradient(135deg,#F5F3FF,#EEF2FF);border:1px solid #C4B5FD;box-shadow:0 4px 12px rgba(124,58,237,0.05);font-family:'Segoe UI',Arial,sans-serif;">
<div style="font-size:19px;font-weight:900;color:#6D28D9;margin-bottom:10px;">🧠 Step 3 · Attention vs RNN Memory</div>
<div style="font-size:14px;line-height:1.8;color:#4F5D63;">
RNNs process a sequence step by step, carrying information from previous time steps through a hidden state, while Attention allows each element to directly consider other elements in the sequence and assign different importance to them. Therefore, RNNs rely on sequential memory to carry context forward, whereas Attention can connect relevant positions directly and process sequence positions in parallel, making it easier to capture relationships between distant elements.
</div>
</div>

<div style="width:100%;box-sizing:border-box;padding:20px 22px;border-radius:15px;background:linear-gradient(135deg,#FFF7ED,#FFFBEB);border:1px solid #FCD34D;box-shadow:0 4px 12px rgba(245,158,11,0.05);font-family:'Segoe UI',Arial,sans-serif;">
<div style="font-size:19px;font-weight:900;color:#B45309;margin-bottom:10px;">🎯 Step 4 · Project Core Architecture</div>
<div style="font-size:14px;line-height:1.8;color:#4F5D63;">
<b style="color:#B45309;">Core Model: LSTM</b>
<br><br>
The LSTM will remain the core architecture for the ECG classification project because the project data is sequential ECG data and the LSTM was successfully trained and evaluated on this task during Day 3. The Transformer concepts learned today provide an alternative architecture for future experimentation, but the current project core remains the LSTM.
</div>
</div>